src/presentation/http/router/noteSettings.ts

```diff
@@ -89,6 +90,34 @@ const NoteSettingsRouter: FastifyPluginCallback<NoteSettingsRouterOptions> = (fa
     return reply.send(noteSettings);
   });
 
+  /**
+   * patch team member role by user and note id
+   */
+  fastify.patch<{
+    Body: {
+      userId: User['id'];
+      noteId: NoteInternalId,
+    }
+    Params: {
+      newRole: keyof typeof MemberRole,
+      },
+    Reply: keyof typeof MemberRole,
+  }>('/new-role/:newRole', {
+    config: {
+      policy: [
+        'authRequired',
+      ],
```

Comment:
leave a todo for adding a policy to check user permission. It should have at least "write" role